README.md

```diff
@@ -1,161 +1,720 @@
+| CID     | Unbounded | A [CID]. Avoiding the identity hash is RECOMMENDED.                                                             |
 
-Since content addressing schemes are backed by cryptographically secure hash functions, their use here prevents forgery of IDs by attackers, and guarantees that CID-based dependencies between tuples will be acyclic.
+Note that there is no primitive `unit` or `null`. Emulating these constructs in data modeling is NOT RECOMMENDED, as they are semantically anemic and alternative modelings are nearly always available. For example, a common use case for using `null` is unset a field. CRDTs SHOULD use history-preserving tombstoning for this purpose instead.
 
-These properties are further leveraged in the design and use of byzantine-fault tolerant CRDTs, as described in [CRDTs](pomo_db/CRDTs.md).
+### 4.1.2 Fact
 
-TODO: Update CRDT link once that info is described somewhere
+The extensional database MUST only store facts as quads (4-tuples) in EAVC[^spoc] format. All fields are REQUIRED, though the causal set MAY be empty.
 
-## 2.5 Provenance Tracking
+[^spoc]: This is [sometimes called SPOC][Hexastore] (🖖) for ["subject-predicate-object-context"][Named Graphs]
 
-TODO: [See notes](https://discord.com/channels/478735028319158273/1033502043656171561/1035339517021921280)
+| Field         | Type          | Description                                        |
+|---------------|---------------|----------------------------------------------------|
+| **E**ntity ID | `Bytes`       | [Entity ID]                                        |
+| **A**ttribute | [`Attribute`] | The name of the value's relationship to the entity |
+| **V**alue     | [`Value`]     | The (primitive) value being associated             |
+| **C**auses    | `Set CID`     | Any direct causal links                            |
 
-## 2.6 Query Engine
+``` typescript
+type Fact = {
+  eid: Bytes;
+  attr: Attribute;
+  val: Value;
+  causal: Set<Cid>;
+}
+```
 
-PomoDB has no specified query language. Instead, an intermediate representation based on the relational algebra, named [PomoRA](pomo_db/pomo_ra.md), is defined.
+The first three fields (entity, attribute, and value) are analogous to a subject-predicate-object statement. For example, "the sky is blue" MAY be represented as $\langle \textsf{skyEID}, \textsf{color}, \textsf{blue} \rangle$.
 
-Implementations MAY define their own user-facing query language, but they are RECOMMENDED to treat PomoRA as a common compilation target for all such languages.
+#### 4.1.2.1 Implicit CID
 
-An OPTIONAL Datalog variant, named [PomoLogic](pomo_db/pomo_logic.md), is also described, along with an OPTIONAL runtime for PomoRA, named [PomoFlow](pomo_db/pomo_flow.md).
+Each tuple within a fact also has an implied [CID][content addressing]. This behaves as an index on all facts. Being derived from the hash of the fact means that the CID can always be rederived.
```

Comment:
> Each tuple within a fact has an implied CID

Do you mean to say:

> Each fact has an implied CID

Reply:
Ah, indeed that's a typo. "Each fact has an implied CID" is correct